### DISCUSSION TITLE
🚨 Alert: GitLocker and Cody Rutscher Exploiting GitHub OAuth Permissions 🚨

### DISCUSSION THREAD

**Discussion by Ab3nXY** [community/community#147750] [Code Security]
### Select Topic Area

General

### Body

Hi GitHub Community,

I want to bring attention to a potential scam targeting developers. A platform called GitLocker, owned by Cody Rutscher, has been posting job listings on Freelancer.com. After applying, they require you to sign up on their website using GitHub.

Here’s the issue: the OAuth permissions they request grant full access to your GitHub account, including private repositories. This raises serious concerns about the security of your code and intellectual property.

Please be cautious and always review OAuth scopes before granting access. If something seems off, don’t proceed. Let’s protect each other by sharing knowledge and staying alert.

Has anyone else encountered similar practices? Let’s discuss and raise awareness!
![Gitlocker](https://github.com/user-attachments/assets/0733c98f-7b86-4dd8-bfd5-e6c9e4628929)

**Comment by codyrutscher** [comment]
GitHub allows for this type of authentication hence why we ASK YOU FIRST. This is the definition of consent. This is slander. Look yourself in the mirror before you point the finger of blame and accusation at someone trying to make an honest living.

**Reply by Ab3nXY** [reply]
https://www.cyclonis.com/remove-gitlocker/

**Reply by Ab3nXY** [reply]
![Screenshot_2024-12-23-14-21-01-09_680d03679600f7af0b4c700c6b270fe7.jpg](https://github.com/user-attachments/assets/fc568d4b-3925-46ca-a06e-5c9fce1f9dd8)

**Comment by codyrutscher** [comment]
You are spreading rumors and disinformation for personal reasons. That type of behavior is aggregious.

**Comment by Akash1134** [comment]
Hi @Ab3nXY , 👋🏻 We really appreciate you flagging this. The best route to get this to the proper GitHub team is to use our abuse reporting tools. Here's all the info:

- Reporting abuse or spam
- Reporting a user
- Reporting an issue or pull request
- Reporting a comment

You can report behavior and content that violates community guidelines and terms. We are going to close this post, but for this and any future incidents, please refer to the links above.

Thank you!

**Reply by Ab3nXY** [reply]
Thank you @Akash1134, In which category should i report this thou?

**Comment by hmz-ak** [comment]
GitLocker is a legitimate platform designed to integrate with GitHub in a secure and transparent manner. The OAuth permissions requested during authentication are used solely to provide users access to their repositories within the application, including both public and private repositories, as intended by GitHub’s API.

**Comment by surafelx** [comment]
" **GitLocker** is an online marketplace where private git repositories are bought and sold. "

### Clarification Regarding GitLocker LLC and the Malicious Group Gitloker

It has come to my attention that there is confusion between **GitLocker LLC** and a malicious entity known as **Gitloker**, as reported on [Malpedia](https://malpedia.caad.fkie.fraunhofer.de/actor/gitloker). I want to clarify that these are entirely unrelated entities. 

#### About GitLocker LLC
GitLocker LLC is a legitimate business registered in Kansas, United States. We operate the website [https://www.gitlocker.com](https://www.gitlocker.com), providing a secure marketplace for developers to **buy and sell private git repositories**. 

Here is an excerpt from the legal terms to provide transparency:

---

We are GitLocker LLC ("Company," "we," "us," "our"), a company registered in Kansas, United States, at **5308 West 139th Terrace, Overland Park, KS 66224-3546**.

We operate the website [https://www.gitlocker.com](https://www.gitlocker.com) (the "Site"), as well as any other related products and services that refer or link to these legal terms (the "Legal Terms") (collectively, the "Services").

**GitLocker** is an online marketplace where private git repositories are bought and sold. 

---

#### Our Commitment to Transparency
We request access to repositories through **proper and clearly defined OAuth permissions**. We ensure that developers maintain control over their intellectual property and are not exposed to any unauthorized access or misuse. 

#### Misuse by Malicious Actors (Gitloker)
The group referred to as **Gitloker** is a malicious entity that has no association with GitLocker LLC. We strongly condemn their actions and any attempt to exploit the trust of developers.

#### Legal Disclaimer
By accessing our Services, users agree to our Legal Terms. This agreement ensures that users are fully informed about the terms of use. If there are updates to our Legal Terms, we notify users by updating the “Last updated” date. The continued use of our Services indicates acceptance of these updates.

---

I urge the community to remain vigilant and differentiate between GitLocker LLC and unrelated malicious actors.

**Reply by AbdullahTRewaa** [reply]
Hi,

I’ve worked with GitLocker and Cody Rutscher for some time now, and I can confidently say that the platform is legitimate. The claims made here are misleading and lack evidence.

The OAuth permissions GitLocker requests are standard for platforms offering integration with GitHub repositories. These permissions are clearly presented during sign-up, allowing users to review and make informed decisions. To date, there has been no evidence of misuse by GitLocker.

Before making such accusations, I encourage verifying the facts and reaching out to the platform for clarification. Spreading unverified claims harms not just businesses but the trust within our community.

Let’s focus on facts and constructive dialogue.

Best,
Abdullah Tahir

**Reply by AbdullahTRewaa** [reply]
The OAuth permissions requested by GitLocker are necessary to simplify the deployment of repositories into the marketplace, ensuring a seamless user experience. These permissions are standard for platforms offering similar functionality, and there is no evidence of misuse.

**Comment by Ab3nXY** [comment]
I will let the proper Github team to figure this out, if giving every gist, access to both private and personal repo as well as full access to user data is mandatory during github Oauth signup.

Note: I created some webapps with social authentication using gmail as well as github so i know what i am talking about.

**Comment by ghostinhershell** [comment]
Hi @Ab3nXY 👋🏻 We really appreciate you flagging this. The best route to get this to the proper GitHub team is to use our [abuse reporting tools](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam). Here's all the info:

- [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam)
- [Reporting a user](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-user)
- [Reporting an issue or pull request](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-an-issue-or-pull-request)
- [Reporting a comment](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-comment)

You can report behavior and content that violates community guidelines and terms. We are going to close this post, but for this and any future incidents, please refer to the links above.

Thank you!
